### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ajardin**

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -13,6 +6,9 @@ BLACKFIRE_SERVER_TOKEN=
 BLACKFIRE_CLIENT_ID=
 BLACKFIRE_CLIENT_TOKEN=
 
-##### PHP
-PHP_IDE_CONFIG=serverName=
-XDEBUG_CONFIG=remote_host= idekey=PHPSTORM remote_port=9001 remote_enable=1 remote_connect_back=0
```
Comment (Fulmenef) [inline .env.dist]: So.. No more Xdebug ?

File: Makefile [hunk 2]
```diff
@@ -1,94 +1,71 @@
-DOCKER          = docker
-DOCKER_COMPOSE  = docker-compose
+SELF_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
+PHP_SERVICE := docker-compose exec php sh -c
```
Comment (Fulmenef) [inline Makefile]: I don't understand why you remove DOCKER_COMPOSE to add PHP_SERVICE. The first one is much more used than the second one.
I think you could have kept the alias DOCKER_COMPOSE and add the PHP_SERVICE alias like `$(DOCKER_COMPOSE) exec php sh -c`

File: Makefile [hunk 3]
```diff
@@ -1,94 +1,71 @@
...
-## ----------------------------------------------------------------------------
-##
+cache: ## Flush cache stored in Redis
+	docker-compose exec redis sh -c "redis-cli -n 1 FLUSHDB"
+	docker-compose exec redis sh -c "redis-cli -n 2 FLUSHDB"
 
-cache: ## Flush everything stored into the "redis" container
-	$(DOCKER_COMPOSE) exec -T redis sh -c "redis-cli FLUSHALL"
+composer: ## Install Composer dependencies from the "php" container
+	$(PHP_SERVICE) "composer install -o --working-dir=$(PROJECT_PATH)"
```
Comment (Fulmenef) [inline Makefile]: It seem to me that PROJECT_PATH is undefined

File: php/Dockerfile [hunk 4]
```diff
@@ -1,68 +1,53 @@
...
+        icu-libs \
+        libjpeg-turbo-dev \
         libmcrypt-dev \
         libpng-dev \
         libxml2-dev \
+        libxml2-utils \
         libxslt-dev \
-        ssmtp && \
+        openssh-client \
+        perl \
```
Comment (Fulmenef) [inline php/Dockerfile]: Wouldn't it be useful to add the git patch dependency here ?

File: Makefile (hunk 2 above)
Comment (ajardin) [reply Makefile]: Long story short:
- `$(DOCKER_COMPOSE)` = 17 characters
- `docker-compose` = 14 characters

From my point of view, there is no need to use a variable for this binary. It's more useful for the `php` service as it's a real shortcut.

File: php/Dockerfile (hunk 4 above)
Comment (ajardin) [reply php/Dockerfile]: Good catch!

File: Makefile (hunk 3 above)
Comment (ajardin) [reply Makefile]: `PROJECT_PATH` as to be defined in the project Makefile as described in the [Wiki](https://github.com/EmakinaFR/docker-magento2/wiki/Makefile). But you're right, I'll add a comment to mention that. 😉

File: Makefile (hunk 2 above)
Comment (Fulmenef) [reply Makefile]: Good for me, thank you for the explanation

File: nginx/servers/magento.conf [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
-    set $MAGE_MODE developer;
-
-    include /etc/nginx/conf.d/magento.conf.sample;
+upstream fastcgi_backend {
+    server  php:9000;
 }
+
+#server {
+#    listen 443 ssl http2;
+#    server_name magento.test;
```
Comment (maxhelias) [inline nginx/servers/magento.conf]: You can add this : `listen [::]:443 ssl http2;` ? 😉

**Pull request merged**